eppo/src/main/java/cloud/eppo/android/EppoClient.java

```diff
@@ -82,219 +113,179 @@ private void refreshConfiguration(InitializationCallback callback) {
-            return null;
-        }
-
-        if (assignmentLogger != null) {
-            String experimentKey = Utils.generateExperimentKey(flagKey, allocationKey);
-            String variationToLog = null;
-            EppoValue typedValue = assignedVariation.getTypedValue();
-            if (typedValue != null) {
-                variationToLog = typedValue.stringValue();
-            }
+        if (assignedValue != null && assignmentLogger != null && evaluationResult.doLog()) {
+            String experimentKey = evaluationResult.getFlagKey();
+            String variationKey = evaluationResult.getVariation().getKey();
+            String allocationKey = evaluationResult.getAllocationKey();
+            Map<String, String> extraLogging = evaluationResult.getExtraLogging();
+            Map<String, String> metaData = new HashMap<>();
+            metaData.put("obfuscated", Boolean.valueOf(isConfigObfuscated).toString());
+            metaData.put("sdkLanguage", "Java (Android)");
+            metaData.put("sdkLibVersion", BuildConfig.EPPO_VERSION);
 
             Assignment assignment = Assignment.createWithCurrentDate(
-                experimentKey,
-                flagKey, 
-                allocationKey, 
-                variationToLog,
-                subjectKey, 
-                subjectAttributes
+                    experimentKey,
+                    flagKey,
+                    allocationKey,
+                    variationKey,
+                    subjectKey,
+                    subjectAttributes,
+                    extraLogging,
+                    metaData
             );
-            assignmentLogger.logAssignment(assignment);
+            try {
+                assignmentLogger.logAssignment(assignment);
+            } catch (Exception e) {
+                Log.w(TAG, "Error logging assignment: "+e.getMessage(), e);
+            }
         }
 
-        return assignedVariation.getTypedValue();
+        return assignedValue != null ? assignedValue : defaultValue;
     }
 
-    public String getAssignment(String subjectKey, String flagKey, SubjectAttributes subjectAttributes) {
-        try {
-            return this.getStringAssignment(subjectKey, flagKey, subjectAttributes);
-        } catch (Exception e) {
-            return throwIfNotGraceful(e);
-        }
+    private boolean valueTypeMatchesExpected(VariationType expectedType, EppoValue value) {
+        boolean typeMatch;
+        switch (expectedType) {
+            case BOOLEAN:
+                typeMatch = value.isBoolean();
+                break;
+            case INTEGER:
+                typeMatch = value.isNumeric()
+                        // Java has no isInteger check so we check value == floor == ceil
+                        && Math.floor(value.doubleValue()) == Math.ceil(value.doubleValue())
+                        && Math.floor(value.doubleValue()) == value.doubleValue();
```

Comment:
Easier to check that `x mod 1 = 0` instead?

Reply:
Ha yes, that would be easier 😅